autogpt_platform/backend/backend/data/block.py

```diff
@@ -492,6 +510,136 @@ def get_blocks() -> dict[str, Type[Block]]:
     return load_all_blocks()
 
 
+def is_block_provider_configured(
+    block_cls: type["Block[BlockSchema, BlockSchema]"],
+) -> bool:
+    """
+    Check if a block has a valid authentication method configured at runtime.
+
+    For example if a block is an OAuth-only block and there env vars are not set,
+    do not show it in the UI.
+
+    """
+    from backend.sdk.registry import AutoRegistry
+
+    # Create an instance to access input_schema
+    try:
+        block = block_cls()
+    except Exception as e:
+        # If we can't create a block instance, assume it's not OAuth-only
+        logger.error(f"Error creating block instance for {block_cls.__name__}: {e}")
+        return True
+    logger.debug(
+        f"Checking if block {block_cls.__name__} has a valid provider configured"
+    )
+
+    # Get all credentials fields from input schema
+    credentials_fields = block.input_schema.get_credentials_fields()
+    if not credentials_fields:
+        logger.debug(
+            f"Block {block_cls.__name__} has no credentials fields - Treating as valid"
+        )
+        return True
+
+    # Check required credentials fields
+    required_fields: set[str] = block.input_schema.get_required_fields()
+    required_credentials = [f for f in credentials_fields if f in required_fields]
+
+    if len(required_credentials) == 0:
+        logger.debug(
+            f"Block {block_cls.__name__} has no required credentials fields - therefore will work without credentials configured"
+        )
+        return True
+
+    if len(required_credentials) > 1:
+        logger.warn(
+            f"Block {block_cls.__name__} has multiple required credentials fields"
+        )
+
+    # Check if theres is any correctly configured credential provider for this block
+    configured_providers = {field_name: [] for field_name in required_credentials}
+
+    for field_name in required_credentials:
+        provider_names = block.input_schema.get_provider_from_credentials_field(
+            field_name
+        )
+        if not provider_names:
+            configured_providers[field_name].append(True)
+            continue
```

Comment:
If this happens, the block has a misconfigured credentials input, which we probably don't want to show in the UI:
```suggestion
        provider_names = block.input_schema.get_provider_from_credentials_field(
            field_name
        )
        if not provider_names:
            logger.warning(
                f"Block {block_cls.__name__} "
                f"has required credentials field {field_name} with no providers"
            )
            return False
```